core/runtime/src/main/java/io/quarkus/runtime/util/ContainerRuntimeUtil.java

```diff
@@ -30,6 +31,20 @@ public static ContainerRuntime detectContainerRuntime() {
         // podman version 2.1.1
         String podmanVersionOutput = getVersionOutputFor(ContainerRuntime.PODMAN);
         boolean podmanAvailable = podmanVersionOutput.startsWith("podman version");
+
+        final String executable = ConfigProvider.getConfig()
+                .getOptionalValue("quarkus.native.container-runtime", String.class).orElse(null);
+        if (executable != null) {
+            if (executable.trim().equalsIgnoreCase("docker") && dockerAvailable) {
+                return ContainerRuntime.DOCKER;
+            } else if (executable.trim().equalsIgnoreCase("podman") && podmanAvailable) {
+                return ContainerRuntime.PODMAN;
+            } else {
+                log.warn("quarkus.native.container-runtime config property must be set to either podman or docker " +
+                        "and the executable must be available. Ignoring it.");
+            }
+        }
+
```

Comment:
This should be revisited after reaching a conclusion for https://github.com/quarkusio/quarkus/pull/31490#discussion_r1121292851